### PR TITLE
Postgis 17

### PR DESCRIPTION
**Pull request by jamonation**
Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [ ] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [ ] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [x] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [x] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [x] The `epoch` field is reset to 0

#### For PRs that add patches

- [x] Patch source is documented

### CODE REVIEW

File: postgis-17.yaml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
-  name: postgis
+  name: postgis-17
   version: 3.5.0
   epoch: 1
   description: Geographic Information Systems Extensions to PostgreSQL
   copyright:
     - license: GPL-2.0-or-later
+  dependencies:
+    provides:
+      - postgis=${{package.version}}
```
Comment (jamonation) [inline postgis-17.yaml:10]:
```suggestion
```

**Review by developer-guy — DISMISSED** [review]

File: postgis-17.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 package:
-  name: postgis
+  name: postgis-17
   version: 3.5.0
   epoch: 1
```
Comment (developer-guy) [inline postgis-17.yaml:4]: when we change the package name here should we reset the epoch?

Comment (jamonation) [inline postgis-17.yaml:4]:
```suggestion
  epoch: 0
```

Comment (jamonation) [reply postgis-17.yaml]: totally, thanks for catching!

**Review by developer-guy — APPROVED** [review]

**Pull request merged**